### ISSUE TITLE
`change:reload` doesn't work as expected when `{q}` is quoted

### ISSUE THREAD

**Issue by sandersantema** [junegunn/fzf#2320]
- [x] I have read through the manual page (`man fzf`)
- [x] I have the latest version of fzf (Although the output of `fzf --version` is somewhat ambiguous namely `0.24 (0.25.0)`)
- [x] I have searched through the existing issues

## Info

- OS
    - [x] Linux
    - [ ] Mac OS X
    - [ ] Windows
    - [ ] Etc.
- Shell
    - [ ] bash
    - [ ] zsh
    - [x] fish
    
## Problem / Steps to reproduce
Although I've encountered this bug in while using fzf.vim I'm fairly certain this is a problem for the whole of fzf.

This is a fairly stripped down function which recreates the problem:
```viml
function! RipgrepFzf(query)
  let spec = {'options': ['--phony', '--query', "", '--bind', 'change:reload:rg "{q}"']}
  call fzf#vim#grep('rg ""', 1, spec, 0)
endfunction
```
When ripgrep is used with a query which is double quoted, search terms inserted in place of `{q}` get single quoted e.g. we've got the query `"{q}"` when we update our search to `foo` we'll get the `'foo'` instead of the expected `foo`. Although in this case having `{q}` surrounded by quotes is completely superfluous it's actually needed when using the regex functionality of ripgrep.

**Comment by junegunn** [comment]
This is not a bug. Any placeholder expression (e.g. `{}`, `{3}`, `{q}`, `{+1}`, ...) is single-quoted so that it can be readily used as an argument in a shell command. (See `--preview` section of the man page). Therefore, you should not put quotes around a placeholder expression.

> needed when using the regex functionality of ripgrep.

Can you give some examples?

**Comment by sandersantema** [comment]
TLDR; quoting `{q}` becomes a problem when it becomes a part of a regex query which is already quoted.

This is the actual function which I had trouble with:
```viml
function! Ripgrep_fzf(query, path, fzfopt, rgopt)
  let escaped_path = shellescape(expand(a:path))
  let reload_bare = 'rg --column --line-number --no-heading --color=always --smart-case ' . a:rgopt

  let regex = ' %s '
  " This is were the problem is (and why it is currently commented):
  " let regex = ' --pcre2 "\[.*\]\((((https?|file)://.*)|.*\.md)\)(*SKIP)(*FAIL)|%s" '
  let reload_fmt = reload_bare . regex . escaped_path
  let initial = printf(reload_fmt, "--pcre2 ''")
  let reload = 'change:reload:' . printf(reload_fmt, '{q}')

  let opt = ['--phony', '--nth=4..', '--query', a:query, '--bind', reload, '--layout=reverse', '--info=inline'] + a:fzfopt
  let spec = {'options': opt}
  call fzf#vim#grep(initial, 1, fzf#vim#with_preview(spec), 0)
endfunction
```

So what's of importance is:
```viml
let regex = ' --pcre2 "\[.*\]\((((https?|file)://.*)|.*\.md)\)(*SKIP)(*FAIL)|%s" '
```

After substitution the regex would become the following:
```
'--pcre2 "\[.*\]\((((https?|file)://.*)|.*\.md)\)(*SKIP)(*FAIL)|{q}" '
```
If you then search for `foo` you get:
```
--pcre2 "\[.*\]\((((https?|file)://.*)|.*\.md)\)(*SKIP)(*FAIL)|'foo'" 
```

Instead of the desired:
```
--pcre2 "\[.*\]\((((https?|file)://.*)|.*\.md)\)(*SKIP)(*FAIL)|foo" 
```

As you can see `{q}` would become part of a regex which is already quoted and needs to be quoted otherwise the regex won't work, this in turn causes the quotes around `{q}` to become part of the regex.

**Comment by sandersantema** [comment]
Bug might have been the wrong term as it's working as intended but causes an error in my use case.

**Comment by junegunn** [comment]
You can easily fix the problem by simply not quoting the placeholder part. Have you tried this?

```vim
let regex = ' --pcre2 "\[.*\]\((((https?|file)://.*)|.*\.md)\)(*SKIP)(*FAIL)|"%s '
```

**Comment by sandersantema** [comment]
That works indeed, thank you! I should've really thought of that. I'll close this issue now.
